demo/portal/src/app/services/wfmService.js

```diff
@@ -0,0 +1,239 @@
+var Promise = require("bluebird");
+var CONSTANTS = require('./constants');
+var _ = require('lodash');
+var shortid = require('shortid');
+
+function WFMApiService(workorderService, workflowService, resultService, userService) {
```

Comment:
Idea was that this can be using sync/api but this service has more logic so I think it's best to extract it.
Doing that now